### PR TITLE
JupyterHub OIDC authentication method

### PR DESCRIPTION
**Pull request by goseind**
This commit provides a Python class called `RucioAuthenticator` extended from JupyterHubs `GenericOAuthenticator` to fetch the access token from the `auth_state` of the user, request a new refresh token, and place it into the spawners environment for the extension to access. It also provides the rest of the helm config for the setup.

ref #20

### CODE REVIEW

**Comment by goseind** [comment]
@bari12 I've drafted a brief guide on how our setup is configured. It should be simple to understand. I plan to create a more detailed guide for the Rucio documentation later on.

Also, could you add @egazzarr and @garciagenrique to this repo?

**Comment by bari12** [comment]
Hi @goseind 
I added @egazzarr and @garciagenrique to the rucio-dev team. For me the doc looks good (Might need some adaption later on once we change some of the token things) - Can you have a look on the failing tests please.

Thanks!

**Review by egazzarr — APPROVED** [review]

File: CONFIGURATION.md [hunk 1]
```diff
@@ -192,10 +192,131 @@ c.IPKernelApp.extensions = ['rucio_jupyterlab.kernels.ipython']
...
+    RUCIO_NAME: "<your-rucio-instance-name>"
+    RUCIO_SITE_NAME: "<your-rucio-instance-site-name>"
+    RUCIO_OIDC_AUTH: "env"
+    RUCIO_OIDC_ENV_NAME: "RUCIO_ACCESS_TOKEN"
+    RUCIO_DEFAULT_AUTH_TYPE: "oidc"
+    RUCIO_OAUTH_ID: "<your-rucio-oauth-id>" # audience
+    RUCIO_DEFAULT_INSTANCE: "<your-rucio-instance-name>""
+    RUCIO_DESTINATION_RSE: "EOS RSE"
+    RUCIO_RSE_MOUNT_PATH: "/eos/eos-rse"
+    RUCIO_PATH_BEGINS_AT: "4"
```
Comment (garciagenrique) [inline CONFIGURATION.md:315]: You could specify that this number makes reference to the amount of child paths in the  `extraVolumen.hostPath.path` argument, that makes reference to the FUSE mounted volumen.

**Review by garciagenrique — APPROVED** [review]

**Comment by garciagenrique** [comment]
Looks fine to me, thanks @goseind for this feature !

File: CONFIGURATION.md (hunk 1 above)
Comment (goseind) [reply CONFIGURATION.md:315]: Good point, will add it! Thanks!

**Comment by bari12** [comment]
Looks good to me too! Thanks everyone. I'll merge it like this. The failing tests in #22 still need to be fixed, but since this is more doc related anyway, it shouldn't really matter.

**Pull request merged**